forge-game/src/main/java/forge/game/ability/SpellAbilityEffect.java

```diff
@@ -585,14 +585,16 @@ protected static void addLeaveBattlefieldReplacement(final Card eff, final Strin
         eff.addReplacementEffect(re);
     }
 
-    // create a basic template for Effect to be used somewhere else
-    public static Card createEffect(final SpellAbility sa, final Player controller, final String name,
-            final String image) {
+    // create a basic template for Effect to be used somewhere els
+    public static Card createEffect(final SpellAbility sa, final Player controller, final String name, final String image) {
+        return createEffect(sa, controller, name, image, controller.getGame().getNextTimestamp());
+    }
+    public static Card createEffect(final SpellAbility sa, final Player controller, final String name, final String image, final long timestamp) {
         final Card hostCard = sa.getHostCard();
         final Game game = hostCard.getGame();
         final Card eff = new Card(game.nextCardId(), game);
 
-        eff.setGameTimestamp(game.getNextTimestamp());
+        eff.setGameTimestamp(timestamp);
         eff.setName(name);
```

Comment:
@tool4ever should we more setSetCode and setRarity be moved to this place too?

from there:
https://github.com/Card-Forge/forge/blob/6d0bf1924248a8188dcec04c2f5382cf7fa45700/forge-game/src/main/java/forge/game/ability/effects/EffectEffect.java#L165-L170

Reply:
yea I think that logic can be default